flowrs-build-designer/rete/flowrs/flowrsNode.ts

```diff
@@ -80,60 +100,92 @@ export class FlowrsNode extends Classic.Node<
                 })
             );
             this.height += 75;
         }
     }
 
-    private setNodeData(data: { [key: string]: any } | null) {
-        if (!data) {
-            return
+    private getConstructorNameToFilterFor(constructorDescription: ConstructorDescription | undefined, typeParameter: string): string | null {
+        if (!constructorDescription?.arguments) {
+            return null;
         }
-
-        this.node_data = JSON.stringify(data, null, 2);
-    }
-
-    private setTypeParameters(typeParameters: { [p: string]: string } | null) {
-        if (!typeParameters) {
-            return;
+        // TODO @mafried: gibts immer nur eine möglichkeit ?
+        let restrictingConstructorType: string | null = null;
+        for (const argument of constructorDescription.arguments) {
+            console.log("Checking for restrictingConstructorType for", typeParameter, "\n", argument)
+            if (argument.type.Generic && argument.type.Generic.name == typeParameter && argument.construction.Constructor) {
+                restrictingConstructorType = argument.construction.Constructor;
+                break;
+            }
         }
+        return restrictingConstructorType;
+    }
 
-        for (const typeParametersKey in typeParameters) {
-            this.genericsResolutionMap.set(typeParametersKey, typeParameters[typeParametersKey]);
+    private chooseMethodAndGetPossibleTypes(constructorNameToFilterFor: null | string, allPossibleTypes: Map<string, TypeDefinition>): [string, TypeDefinition][] {
+        if (constructorNameToFilterFor) {
+            let possibleTypes = this.getFilteredTypesList(constructorNameToFilterFor, allPossibleTypes);
+            console.log("RestrictingConstructorType resulted into filtered list", possibleTypes);
+            return possibleTypes;
+        } else {
+            let possibleTypes = Array.from(allPossibleTypes.entries());
+            console.log("Full list available", possibleTypes);
+            return possibleTypes;
         }
-
-        console.log("Type parameters set:", this.genericsResolutionMap);
     }
 
-    private calculateCompatibleTypes(constructorDescription: ConstructorDescription | undefined, allPossibleTypes: Map<string, TypeDefinition>) {
-        if (!constructorDescription?.arguments) {
-            return;
-        }
-        console.log("arguments", constructorDescription.arguments)
-        for (const argument of constructorDescription.arguments) {
-            console.log("check if fitting:", argument.type.Generic, argument.construction.Constructor)
-            if (argument.type.Generic && argument.construction.Constructor) {
-                this.compatibleTypes.set(argument.type.Generic.name, this.filterTypesWithConstructor(argument.construction.Constructor, allPossibleTypes))
+    // TODO @mafried: Wie soll ich aus dem TimerNode Fall rausfiltern, das für T ein Debug Node ein gültiger Timer sein kann?
```

Comment:
Momentan werden Rust where-Klauseln nicht im JSON reflektiert (ebenso Traits, die ein Typ implementiert). Sie haben also an der Stelle (noch) keine Möglichkeit herauszufinden, welche Type-Constraints auf T liegen. Im Rust Code sieht's für den TimerNode so aus: pub struct TimerNode<T, U> where T: TimerStrategy<U>. Timer müssen also TimerStrategy implementieren. Es bleibt uns hier also nur übrig, alles zuzulassen und dann meckert erst der Compiler beim Bauen (das ist einer der Gründe, warum detaillierte Compiler-Fehler in der GUI angezeigt werden sollten).